src/libsystemd-network/sd-dhcp6-lease.c

```diff
@@ -445,6 +445,37 @@ int sd_dhcp6_lease_get_fqdn(sd_dhcp6_lease *lease, const char **ret) {
         return 0;
 }
 
+int dhcp6_lease_set_captive_portal(sd_dhcp6_lease *lease, const uint8_t *optval, size_t optlen) {
+        _cleanup_free_ char *captive_portal = NULL;
+        int r;
+
+        assert(lease);
+        assert(optval || optlen == 0);
+
+        if (optlen == 0) {
+                mfree(lease->captive_portal);
+                return 0;
+        }
+
+        r = make_cstring((char *)optval, optlen, MAKE_CSTRING_REFUSE_TRAILING_NUL, &captive_portal);
+        if (r < 0)
+                return r;
+
+        free_and_replace(lease->captive_portal, captive_portal);
+        return 0;
```

Comment:
why not use dhcp_option_parse_string() here?

Reply:
My intention was to refuse a trailing NUL byte in this case, as I don't think the comment in dhcp_option_parse_string applies to dhcp6? Not sure though. Even if it doesn't might be prudent to create a dhcp6_option_parse_string...

Reply:
I've gone ahead and added such a function.